pmultiqc/modules/quantms/quantms.py

```
@@ -1838,6 +2015,261 @@ def getSpectrumCountAcrossRep(condition_count_dict: dict):
+                            "accession_group", "experimentalMassToCharge", "calculatedMassToCharge",
+                            "isDecoy", "pep_to_prot_unique", "search_engine_score", "stand_spectra_ref",
+                            "location", "filename", "retention_time"]].drop_duplicates().reset_index()
+        else:
+            psm = mzid_table[["spectrumID", "PeptideSequence", "chargeState", "Modifications",
+                            "accession_group", "experimentalMassToCharge", "calculatedMassToCharge",
+                            "isDecoy", "pep_to_prot_unique", "search_engine_score", "stand_spectra_ref",
+                            "location", "filename"]].drop_duplicates().reset_index()
+        
+        for m, group in psm.groupby("filename"):
+            self.oversampling_plot = Histogram('MS/MS counts per 3D-peak', plot_category='frequency', breaks=[1, 2, 3])
+            for i, j in group.groupby(["PeptideSequence", "chargeState", "Modifications"]):
```

Ruff

Loop control variable `i` not used within loop body

(B007)